src/bin/tester.rs

```diff
@@ -136,7 +175,88 @@ fn ambiguity_check(handle: ModuleContentsHandle) -> Result<(), MoreThanOne> {
     Ok(())
 }
 
-fn main() {
+#[derive(Debug, Default, Add)]
+struct Counters {
+    total_count: u16,
+    unambiguous_count: u16,
+    ambiguous_count: u16,
+    too_short_count: u16,
+    no_parse_count: u16,
+}
+
+#[derive(Debug)]
+enum ParseResult {
+    Unambiguous,
+    Ambiguous,
+    Partial,
+    Error,
+}
+
+impl ParseResult {
+    fn compact_display(&self) -> &'static str {
+        match self {
+            ParseResult::Unambiguous => "-",
+            ParseResult::Ambiguous => ".",
+            ParseResult::Partial => "X",
+            ParseResult::Error => "L",
+        }
+    }
+}
+
+fn process(file: walkdir::DirEntry, verbose: bool) -> ParseResult {
+    let mut stdout = io::stdout();
+    let path = file.into_path();
+
+    let out = parse_file_with(&path, |result| {
+        let mut ambiguity_result = Ok(());
+        let start = Instant::now();
```

Comment:
This is after the parse, it only measures `ambiguity_check` (which is a graph traversal, pretty insignificant)